### PR TITLE
strip file times to make tarballs more reproducible

### PR DESCRIPTION
**Pull request by mxey**

### CODE REVIEW

**Comment by mxey** [comment]
I also think it would be sensible to strip all file modes except for executable, similar to Git. Should I add this?

**Comment by jdolitsky** [comment]
@mxey sure that sounds good.. @shizhMSFT any thoughts on this?

**Comment by shizhMSFT** [comment]
@mxey @jdolitsky File times are useful for many scenarios, especially for the scenario that we can use the uploaded directory (tar.gz) as a layer of a real image. Thus we should not strip file times.

**Comment by mxey** [comment]
> File times are useful for many scenarios, especially for the scenario that we can use the uploaded directory (tar.gz) as a layer of a real image. Thus we should not strip file times.

@shizhMSFT the same could be said for UID and GID, which you already strip

**Comment by shizhMSFT** [comment]
It is not stripping UID and GID but `chown` to `root`.

BTW, the scenario I have mentioned above can be found at https://github.com/sajayantony/dotnet-oci

**Comment by jdolitsky** [comment]
Perhaps this is something we can pass as feature flag on push if necessary?

**Comment by mxey** [comment]
> It is not stripping UID and GID but chown to root.

OK, in the same sense, I am not stripping times, I am just setting them to epoch 0

**Comment by mxey** [comment]
> Perhaps this is something we can pass as feature flag on push if necessary?

That would work for me.

**Comment by jzelinskie** [comment]
We definitely should have parameters that clarify the manipulation of the tar headers. Different use cases will definitely have different requirements and may want the same tarball with different file attributes to be unique.

It's best to have an opinion, but leave a backdoor. Make the easy things easy, and the hard things possible.

**Comment by mxey** [comment]
@jzelinskie So should I modify my pull request to make this into an option?

**Comment by jzelinskie** [comment]
I think so. What say you @jdolitsky?

**Comment by jdolitsky** [comment]
@mxey @jzelinskie sorry for delay - yes that sounds good

File: pkg/content/file.go [hunk 1]
```diff
@@ -28,6 +28,9 @@ type FileStore struct {
 	DisableOverwrite          bool
 	AllowPathTraversalOnWrite bool
 
+	// Reproducible enables stripping times from added files
+	Reproducible bool
```
Comment (shizhMSFT) [inline pkg/content/file.go]: Maybe `StripTimes` is better?

Comment (mxey) [reply pkg/content/file.go]: @shizhMSFT I decided to make it more generic, in case there are more things to strip

**Pull request merged**